experimental/traffic-portal/src/app/api/testing/user.service.ts

```diff
@@ -428,6 +438,50 @@ export class UserService {
 		return this.roles;
 	}
 
+	/**
+	 * Creates a new role.
+	 *
+	 * @param role The role to create.
+	 * @returns The created role.
+	 */
+	public async createRole(role: RequestRole): Promise<ResponseRole> {
+		const resp = {
+			...role,
+		};
+		this.roleDetail.push(resp);
+		return resp;
+	}
+
+	/**
+	 * Updates an existing role.
+	 *
+	 * @param role The role to update.
+	 * @returns The updated role.
+	 */
+	public async updateRole(role: ResponseRole): Promise<ResponseRole> {
+		const roleName = this.roleDetail.findIndex(r => r.name === role.name);
+		if (roleName < 0 ) {
+			throw new Error(`no such Role: ${role.name}`);
+		}
+		this.roleDetail[roleName] = role;
+		return role;
+	}
+
+	/**
+	 * Deletes an existing role.
+	 *
+	 * @param role The role to be deleted.
+	 * @returns The deleted role.
+	 */
+	public async deleteRole(role: string | ResponseRole): Promise<ResponseRole> {
```

Comment:
The call signature of this testing method is not compatible with the "concrete" service; according to the documentation, your concrete service method is correct and this should return `Promise<void>`.

Reply:
Nope, I changed the "concrete" service to return ```Promise<ResponseRole>```